## Assignment 7 - Regression

#### Task 1
- Import any libraries will be needed for this assignment

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import statsmodels. formula.api as smf

#### Task 2
- Open the file _usedcars.csv_ using Pandas and store in a dataframe variable called _cars_
- Print the first 10 rows of the dataframe to the notebook

In [2]:
cars = pd.read_csv('usedcars.csv')
cars.head(10)

year  mileage transmission  transmission_code  price
0  2011     7413         AUTO                  1  21992
1  2011    10926         AUTO                  1  20995
2  2011     7351         AUTO                  1  19995
3  2011    11613         AUTO                  1  17809
4  2012     8367         AUTO                  1  17500
5  2010    25125         AUTO                  1  17495
6  2011    27393         AUTO                  1  17000
7  2010    21026         AUTO                  1  16995
8  2011    32655         AUTO                  1  16995
9  2010    36116         AUTO                  1  16995

#### Task 3
- Define the regression model and store the result in a variable called _model_.  The regression model should predict price by using year, mileage, and transmission_code as predictors. You should use the _cars_ dataframe as the data source for this analysis.
- Fit the model and store the result in a variable called _result_.

In [3]:

model = smf.ols('price ~ year + mileage + transmission_code', data=cars)
result = model.fit()


#### Task 4
- Print the result summary

In [4]:

print(result.summary())


                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.786
Model:                            OLS   Adj. R-squared:                  0.782
Method:                 Least Squares   F-statistic:                     179.0
Date:                Sun, 20 Apr 2025   Prob (F-statistic):           1.04e-48
Time:                        06:41:22   Log-Likelihood:                -1303.6
No. Observations:                 150   AIC:                             2615.
Df Residuals:                     146   BIC:                             2627.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept         -1.541e+06   1.68e+0

#### Task 5

In the markdown cell below, answer the following questions in the cell:

1) What is the percent of variance explained by this regression model?

2) Is the regression model statistically significant overall at an alpha of 0.05?  How can you tell?

3) Are any of the coefficients __NOT__ statistically significant at an alpha of 0.05?  How can you tell?

4) How much will the price increase (or decrease) for each mile that the car has driven?

5) We coded the Manual transmission as 0 and Automatic as 1. Compared to a car with manual transmission, how much more (or less) expensive is a cart with automatic transmission?

#### Answers

1) The percent of variance explained by this regression model is 78.6%

2) Yes the model is statistically significant.The p-value for Prob (F-statistic)the p-value of 1.04e-48, which is less than 0.05. This shows that the model as a whole is a good fit.

3) All the predictors have p-values less than 0.05, meaning all are statistically significant.

4) The coefficient for mileage is -0.0453, meaning that for every additional mile, the price decreases by about $0.045.

5) The coefficient for transmission_code is 821.70, so a car with automatic transmission is, on average, $821.70 more expensive than a similar car with manual transmission.

